src/test/ui/builtin-superkinds/builtin-superkinds-in-metadata.stderr

```diff
@@ -10,10 +10,9 @@ note: required because it appears within the type `X<T>`
 LL | struct X<T>(T);
    |        ^
 note: required by a bound in `RequiresRequiresShareAndSend`
-  --> $DIR/auxiliary/trait_superkinds_in_metadata.rs:7:58
+  --> /rustc/FAKE_PREFIX/src/test/ui/builtin-superkinds/auxiliary/trait_superkinds_in_metadata.rs:7:58
    |
-LL | pub trait RequiresRequiresShareAndSend : RequiresShare + Send { }
-   |                                                          ^^^^ required by this bound in `RequiresRequiresShareAndSend`
+   = note: required by this bound in `RequiresRequiresShareAndSend`
```

Comment:
It seems like we should special case this in note_obligation_cause_blah, and make it handle it itself, to avoid redundant text.

Reply:
With https://github.com/rust-lang/rust/pull/105514 merged, this won't be an issue.